metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/parser/AnalysisFileTypeFromSet.java

```diff
@@ -0,0 +1,157 @@
+        List<AnalysisFileType> subAnalysisFileList = new ArrayList<>();
+
+        AnalysisType[] analysisType = analysisSet.getANALYSISArray();
+        for (AnalysisType analysis : analysisType) {
+            AnalysisType.FILES analysisFILES = analysis.getFILES();
+            AnalysisFileType[] analysisFile = analysisFILES.getFILEArray();
+            for (AnalysisFileType anaFile : analysisFile) {
+                subAnalysisFileList.add(anaFile);
+            }
+        }
+        return subAnalysisFileList;
+    }
+
+    public AnalysisSetType getAnalysisSet(String xmlStr) throws XmlException {
+        return ANALYSISSETDocument.Factory.parse(xmlStr).getANALYSISSET();
+    }
+
+    public List<AnalysisFileType> extractFromSqlXml(SQLXML sqlXml) {
+        String xmlStr;
+        List<AnalysisFileType> analysisFileList = new ArrayList<>();
+
+        try {
+            xmlStr = sqlXml.getString();
+            AnalysisSetType analysisSet;
+            try {
+                analysisSet = getAnalysisSet(xmlStr);
+                analysisFileList = extract(analysisSet);
+            } catch (XmlException e) {
+                logger.info("Unable to convert XML String to AnalysisSet: ", xmlStr);
+            }
+        } catch (SQLException e) {
+            logger.info("Unable to convert SQLXML Object to String: ", sqlXml.toString());
+        }
+        return analysisFileList;
+    }
+
+    private Properties loadPropertiesFile() throws IOException {
+        Properties properties = new Properties();
+        InputStream inFile = null;
+        try {
+            inFile = new FileInputStream(propertiesFile);
+            try {
+                properties.load(inFile);
+            } catch (IOException e) {
+                logger.info("Unable to load JDBC connection properties file");
+                throw e;
+            }
+        } catch (FileNotFoundException e) {
+            logger.info("JDBC connection properties file not found: " + propertiesFile);
+            throw e;
+        } finally {
+            if (inFile != null) {
+                inFile.close();
+            }
+        }
+        return properties;
+    }
+
+    private String createUrl(Properties appProperties) {
+        String url = appProperties.getProperty("OracleJDBC.url");
+        String host = appProperties.getProperty("OracleJDBC.host");
+        String protocol = appProperties.getProperty("OracleJDBC.protocol");
+        String serviceName = appProperties.getProperty("OracleJDBC.servicename");
+        String port = appProperties.getProperty("OracleJDBC.port");
+        url = url + ":@(DESCRIPTION =(ADDRESS_LIST =(ADDRESS = (PROTOCOL = " + protocol + ")(HOST = " + host + ")(PORT = " +
+                port + ")))(CONNECT_DATA =(SERVER = DEDICATED)(SERVICE_NAME = " + serviceName + ")))";
+
+        return url;
+    }
+
+    public List<AnalysisFileType> getAnalysisFileObject() throws IOException, SQLException {
```

Comment:
The class name is `AnalysisFileTypeFromSet` and it implements `TypeFromSet` interface, while this method is about from connecting to database all the way through to getting the list of `AnalysisFileType` objects. I think this class has too many more responsibilities than its name suggests and a lot of logics here are not re-usable for other object types where they should be. Please try to abstract those common logics to re-usable components and re-design the interfaces and their implementations.